### PR TITLE
F #3440: Fix bug creating NSX networks

### PR DESCRIPTION
**Pull request by mouyaq**
Fix bug when creating NSX network without
transport zone. Added more checks and
exceptions to manage them.

### CODE REVIEW

File: share/hooks/vcenter/create_vcenter_net.rb [hunk 1]
```diff
@@ -158,8 +158,13 @@ def err_and_exit(error_message)
                 <controlPlaneMode>UNICAST_MODE</controlPlaneMode>\
                 <guestVlanAllowed>false</guestVlanAllowed>\
             </virtualWireCreateSpec>"
-        logical_switch = NSXDriver::VirtualWire
-                         .new(nsx_client, nil, tz_id, virtual_wire_spec)
+        begin
+            logical_switch = NSXDriver::VirtualWire
+                             .new(nsx_client, nil, tz_id, virtual_wire_spec)
+        rescue StandardError => e
+            STDERR.puts "ERROR: #{e.message}"
```
Comment (tinova) [inline share/hooks/vcenter/create_vcenter_net.rb]: This should also have configurable backtrace:

    if VCenterDriver::CONFIG[:debug_information]
        STDERR.puts "#{message} #{e.backtrace}"
    end

File: share/hooks/vcenter/create_vcenter_net.rb [hunk 2]
```diff
@@ -180,9 +185,14 @@ def err_and_exit(error_message)
...
             }
         )
-        logical_switch = NSXDriver::OpaqueNetwork
-                         .new(nsx_client, nil, nil, opaque_network_spec)
-        # Get NSX_VNI
+        begin
+            logical_switch = NSXDriver::OpaqueNetwork
+                             .new(nsx_client, nil, tz_id, opaque_network_spec)
+        rescue StandardError => e
+            STDERR.puts "ERROR: #{e.message}"
```
Comment (tinova) [inline share/hooks/vcenter/create_vcenter_net.rb]: same here

    if VCenterDriver::CONFIG[:debug_information]
        STDERR.puts "#{message} #{e.backtrace}"
    end

File: share/hooks/vcenter/create_vcenter_net.rb (hunk 1 above)
Comment (mouyaq) [reply share/hooks/vcenter/create_vcenter_net.rb]: Done

File: share/hooks/vcenter/create_vcenter_net.rb (hunk 2 above)
Comment (mouyaq) [reply share/hooks/vcenter/create_vcenter_net.rb]: Done

**Pull request merged**